src/MSABasic.sol

```diff
@@ -127,6 +129,9 @@ contract MSABasic is IMSA, ExecutionHelper, ModuleManager {
         payable
         onlyEntryPointOrSelf
     {
+
+        if (!IModule(module).isModuleType(moduleTypeId)) revert MismatchModuleTypeId(moduleTypeId);
```

Comment:
maybe we should check if module passed supports interfaceId for IModule?
@kopy-kat

Reply:
`isModuleType` should be sufficient right?

Reply:
yes it is sufficient. we should write more tests though